drivers/flash/CMakeLists.txt

```diff
@@ -18,11 +18,19 @@ zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_NIOS2_QSPI soc_flash_nios2_qspi.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_GECKO flash_gecko.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_RV32M1 soc_flash_rv32m1.c)
 zephyr_library_sources_ifdef(CONFIG_FLASH_STM32_QSPI flash_stm32_qspi.c)
-zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_NOR flash_mcux_flexspi_nor.c)
+if(CONFIG_SOC_SERIES_IMX_RT6XX)
+	zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_NOR flash_mcux_flexspi_mx25um51345g.c)
+else()
+	zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_NOR flash_mcux_flexspi_nor.c)
+endif()
```

Comment:
Please add a new config to `Kconfig.mcux` to make this:
```
zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_MX25UM51345G flash_mcux_flexspi_mx25um51345g.c)
```

And have the board enable this new config instead of CONFIG_FLASH_MCUX_FLEXSPI_NOR

That will allow you to remove the SoC-specific conditional here.

Reply:
If I do that, I would have to add another config. based driver files inclusion in same CMakeLists.txt. Let me know if something as follows is ok to be added. Because this would duplicate existing `CONFIG_FLASH_MCUX_FLEXSPI_NOR` based option in some way.
`zephyr_include_directories_ifdef(
  CONFIG_FLASH_MCUX_FLEXSPI_MX25UM51345G
  ${ZEPHYR_BASE}/drivers/memc
)`

Reply:
Changes are pushed. Please check.